source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -666,4 +744,93 @@ namespace AdaptiveCards::Rendering::Uwp
         auto foundSvg = url.find("svg");
         return !(foundSvg == std::string::npos);
     }
+
+    void render_xaml::XamlBuilder::ParseXmlForHeightAndWidth(winrt::XmlDocument const& xmlDoc,
+                                                             winrt::SvgImageSource const& imageSource)
+    {
+        if (xmlDoc)
+        {
+            auto rootElement = xmlDoc.DocumentElement();
+            auto height = rootElement.GetAttribute(L"height");
+            auto width = rootElement.GetAttribute(L"width");
+
+            // We only need to set height or width, not both (fixes aspect ratio for person style)
+            bool isHeightSet = false;
+
+            if (!height.empty())
+            {
+                if (auto heightAsDouble = HStringToDouble(height))
```

Comment:
>heightAsDouble [](http://example.com/codeflow?start=25&length=14)

Seems strange to put a `double` in an if block.

Reply:
Since height & width can be a string value (ex: auto), I wanted to make sure a valid double was returned.

With the switch to `std::stod`, I'll switch to a try/catch block

Reply:
Perhaps you could change `HStringToDouble` to `TryHStringToDouble` and it returns a `std::optional<double>`.  That way it has an easy way to express that it has nothing to return.